# Prevents OpenGL types from being defined in gl3.h if MKXPZ_NO_GL_TYPES is defined.

--- a/api/GLES3/gl3.h
+++ b/api/GLES3/gl3.h
@@ -40,13 +40,16 @@ extern "C" {
 #ifndef GL_ES_VERSION_2_0
 #define GL_ES_VERSION_2_0 1
 #include <KHR/khrplatform.h>
+#ifndef MKXPZ_NO_GL_TYPES
 typedef khronos_int8_t GLbyte;
 typedef khronos_float_t GLclampf;
 typedef khronos_int32_t GLfixed;
 typedef khronos_int16_t GLshort;
 typedef khronos_uint16_t GLushort;
 typedef void GLvoid;
+#endif // MKXPZ_NO_GL_TYPES
 typedef struct __GLsync *GLsync;
+#ifndef MKXPZ_NO_GL_TYPES
 typedef khronos_int64_t GLint64;
 typedef khronos_uint64_t GLuint64;
 typedef unsigned int GLenum;
@@ -60,6 +63,7 @@ typedef int GLint;
 typedef unsigned char GLboolean;
 typedef int GLsizei;
 typedef khronos_uint8_t GLubyte;
+#endif // MKXPZ_NO_GL_TYPES
 #define GL_DEPTH_BUFFER_BIT               0x00000100
 #define GL_STENCIL_BUFFER_BIT             0x00000400
 #define GL_COLOR_BUFFER_BIT               0x00004000
@@ -651,7 +655,9 @@ GL_APICALL void GL_APIENTRY glViewport (GLint x, GLint y, GLsizei width, GLsizei
 
 #ifndef GL_ES_VERSION_3_0
 #define GL_ES_VERSION_3_0 1
+#ifndef MKXPZ_NO_GL_TYPES
 typedef khronos_uint16_t GLhalf;
+#endif // MKXPZ_NO_GL_TYPES
 #define GL_READ_BUFFER                    0x0C02
 #define GL_UNPACK_ROW_LENGTH              0x0CF2
 #define GL_UNPACK_SKIP_ROWS               0x0CF3
